File: in_Java/0724 Find Pivot Index.java
class Solution {
    public int pivotIndex(int[] nums) {
        if(nums == null || nums.length == 0) return -1;
        int[] a = new int[nums.length];
        int[] b = new int[nums.length];
        a[0] = nums[0];
        b[nums.length-1] = nums[nums.length-1];
        for(int i = 1; i < nums.length; i++){
            a[i] = nums[i] + a[i-1];
            b[nums.length-1-i] = nums[nums.length-1-i] + b[nums.length-i];
        }
        for(int i = 0; i < nums.length; i++){
            if(a[i] == b[i]) return i;
        }
        return -1;
    }
}

File: in_Python/0724 Find Pivot Index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        leftsum = []
        rightsum = []
        if nums == None or len(nums) == 0:
            return -1
        leftsum.append(nums[0])
        rightsum.append(nums[-1])
        for i in range(1, len(nums)):
            leftsum.append(leftsum[i-1] + nums[i])
            rightsum.append(rightsum[i-1] + nums[len(nums)-1-i])
        for i in range(len(nums)):
            if leftsum[i] == rightsum[len(nums)-1-i]:
                return i
        return -1